src/shared/codelens/typescriptCodeLensProvider.ts

```diff
@@ -37,6 +40,54 @@ async function getSamProjectDirPathForFile(filepath: string): Promise<string> {
     return path.dirname(packageJsonPath)
 }
 
+export async function invokeLambda(params: LambdaLocalInvokeParams
+        & {
+            runtime: string,
+            settings: SettingsConfiguration,
+            processInvoker: SamCliProcessInvoker,
+            localInvokeCommand: SamLocalInvokeCommand,
+            telemetryService: TelemetryService,
+            outputChannel: vscode.OutputChannel,
```

Comment:
Can/should some of these be added to LambdaLocalInvokeParams?

Reply:
I'm hoping to remove most of these intermediate state-holders (such as `LambdaLocalInvokeParams`).  There should be _one_ data structure shared for "launch", not dozens of half-baked ones.